src/middlewared/middlewared/plugins/snapshot.py

```diff
@@ -0,0 +1,135 @@
+from datetime import time
+
+from middlewared.schema import accepts, Bool, Cron, Dict, Int, Patch, Str
+from middlewared.service import CRUDService, private, ValidationErrors
+from middlewared.validators import Time
+
+
+class PeriodicSnapshotService(CRUDService):
+
+    class Config:
+        datastore = 'storage.task'
+        datastore_prefix = 'task_'
+        datastore_extend = 'pool.snapshot.periodic_snapshot_extend'
+        namespace = 'pool.snapshot'
+
+    @private
+    def periodic_snapshot_extend(self, data):
+        data['begin'] = str(data['begin'])
+        data['end'] = str(data['end'])
+        data['repeat_unit'] = data['repeat_unit'].upper()
+        data['ret_unit'] = data['ret_unit'].upper()
+        data['schedule'] = data.pop('byweekday')
+        return data
+
+    @private
+    async def common_validation(self, data, schema_name):
+        verrors = ValidationErrors()
+
+        interval_choices = [5, 10, 15, 30, 60, 120, 180, 240, 360, 720, 1440, 10080, 20160, 40320]
```

Comment:
Seems like we have a finite number that could be part of an enum? Could you investigating about possibility of adding EnumMixin to Int?